File: 3447-clear-digits/3447-clear-digits.java
class Solution {
    public String clearDigits(String s) {
        char [] ans = new char[s.length()+1];
        int x = 0 ;
        for(int i=0 ;i<s.length();i++){
            if(s.charAt(i)>96&&s.charAt(i)<127){
                ans[x++] = s.charAt(i);
            }else { 
                if(x>0){
                    x-- ;
                }
            } 
        }
        return new String(Arrays.copyOfRange(ans,0,x)) ;
    }
}

File: 3447-clear-digits/3447-clear-digits.c
char* clearDigits(char* s) {
    int n = strlen(s);
    char *ans  = (char*)malloc(sizeof(char)*(n+1)) ;
    int x = 0 ;
    for(int i=0 ;i<n;i++){
        if((s[i]>96&&s[i]<127)){
            ans[x++] = s[i] ;
        }else {
            if(x>0)
            x-- ;
        } 
    }
    ans[x] = 0 ;
    return ans ;
}